src/HealthChecks.MongoDb/MongoDbHealthCheck.cs

```diff
@@ -46,10 +48,30 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 // this you can check a specified database.
                 // Related with issue #43 and #617
 
-                await mongoClient
-                    .GetDatabase(_specifiedDatabase)
-                    .RunCommandAsync(_command, cancellationToken: cancellationToken)
-                    .ConfigureAwait(false);
+                for (int attempt = 1; attempt <= MAX_PING_ATTEMPTS; attempt++)

```

Comment:
Should we consider configure a delay between retry and the max retry count

Reply:
@WeihanLi 

I have been thinking about this.

Seems adding a delay between retries won't give any benefit.

If a connection attempt fails due to networking problems, the operation will take a long time to fail. Practically - could take 20-30 seconds, I guess it depends on Mongo client timeout settings, which we don't know.

Let's examine a simple case: MAX_PING_ATTEMPTS = 3, and Mongo client has a timeout of 30 seconds. In this case single check will take ~90 seconds. A second attempt is made after ~30 seconds, and the last attempt is made after ~1minute.

​If a check failed due to permissions/missing collection, etc... - delay between the checks does not have any benefit.